service/storage_proxy.cc

```diff
@@ -5260,6 +5260,45 @@ const db::hints::host_filter& storage_proxy::get_hints_host_filter() const {
     return _hints_manager.get_host_filter();
 }
 
+future<> storage_proxy::create_hint_queue_sync_point(utils::UUID sync_point_id, std::vector<gms::inet_address> endpoints, clock_type::time_point deadline) {
```

Comment:
Suppose the future returned here completes when the sync point is reached? Wouldn't that be easier to use? Instead of creating and storing sync_point_id, create the future using this function and store it.

Reply:
Also, the feeling that this belongs in commitlog and should use replay_position is strengthened.

Reply:
A hint sync point created in storage_proxy can wait for multiple hint queues. It wouldn't make sense to use replay_position here because for each hint queue it will be different.